instrumentation/jdbc/src/main/java/com/solarwinds/opentelemetry/instrumentation/StatementTruncator.java

```diff
@@ -55,6 +55,14 @@ public static void maybeTruncateStatement(Context context) {
       int sqlMaxLength =
           ConfigManager.getConfigOptional(
               ConfigProperty.AGENT_SQL_QUERY_MAX_LENGTH, DEFAULT_SQL_MAX_LENGTH);
+
+      if (sql.contains("traceparent")) {
+        // Remove trace context from captured db.statement. For hibernate, we inject context before
+        // it's captured by upstream.
+        sql = sql.replaceAll("/.*traceparent.*/", "");
+        span.setAttribute(SemanticAttributes.DB_STATEMENT, sql);
+      }
```

Comment:
it's actually fine if we capture the injected comment as part of `db.statement` instead of a separate `QueryTag` attr.

Reply:
that increases the query length and could trigger truncation in the subsequent blocks which could after fingerprinting.